modules/app/src/main/resources/lib/collaboration.js

```diff
@@ -1,33 +1,133 @@
 /* global __ */
 
+const sharedMemoryLib = require('/lib/xp/grid');
+const eventLib = require('/lib/xp/event');
+
+const TTL_SECONDS = 2 * 60 * 1000;
+
+const SHARED_MAP = sharedMemoryLib.getMap('contentstudio.collaboration.contents');
+
 function required(params, name) {
     const value = params[name];
-    if (value === undefined) {
+    if (typeof value === 'undefined' || value === null) {
         throw `Parameter "${name}" is required`;
     }
-    return value;
+}
+
+function CollaboratorsModifier(params) {
+    this.collaboratorKey = `${params.sessionId}=${params.userKey}`;
+    this.contentId = params.contentId;
+    this.collaborators = [];
+
+    this.removeExpired = function (timeAt) {
+        const self = this;
+        const originalSize = this.collaborators.length;
+        this.collaborators = this.collaborators.filter(collaborator => {
+            let lastJoin = self.extractLastJoin(collaborator);
+            return timeAt - lastJoin < TTL_SECONDS;
+        });
+        return originalSize > this.collaborators.length;
+    };
+
+    this.removeCurrent = function () {
```

Comment:
this function is confusing - it not only removes current collaborator, but also returns something, this should not be like this. Split it into 2 methods: `removeCurrent()` should just filter out current user, `isCurrentUserRemoved()` should be used further in the code and it must use `Array.some()` to determine if user is there or not
However in some cases this is appropriate, so you may leave it as it is
Same issue with `removeExpired()`

Reply:
I will leave it as is. Typically remove method returns true/false or previous (removed) value.